### PR TITLE
Create new example which showcases Route usage.

### LINKED ISSUE #60: Create new example which showcases Route usage
New example can be added to `navigation-ios-examples` which will allow to showcase how to:
- initialize a Route in code (instead of direct usage of MapboxDirections API)
- deserialize Route from JSON file

### PR DESCRIPTION
**Pull request by MaximAlien**
Resolves #60.

This PR depends on changes made in Directions repo. PR [here](https://github.com/mapbox/mapbox-directions-swift/pull/430).

### CODE REVIEW

File: Navigation-Examples/Constants.swift [hunk 1]
```diff
@@ -45,13 +49,6 @@ let listOfExamples: [NamedController] = [
         storyboard: nil,
         pushExampleToViewController: false
     ),
-    (
```
Comment (MaximAlien) [inline Navigation-Examples/Constants.swift:48]: For some reason AdvancedViewController was added twice. Let me know if that's okay to keep only one.

File: Navigation-Examples.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -23,7 +26,6 @@
 		B40ADDC9ABB7E7F6703997CA /* Pods_DocsCode.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 9B796D93B6E7934BFD326FA8 /* Pods_DocsCode.framework */; };
 		C52BAA782040D5030086EC6B /* Custom-Destination-Marker.swift in Sources */ = {isa = PBXBuildFile; fileRef = C52BAA772040D5030086EC6B /* Custom-Destination-Marker.swift */; };
 		C58FB85A1FE899B800C4B491 /* AppDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = C58FB8591FE899B800C4B491 /* AppDelegate.swift */; };
-		C58FB85C1FE899B800C4B491 /* ViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = C58FB85B1FE899B800C4B491 /* ViewController.swift */; };
```
Comment (MaximAlien) [inline Navigation-Examples.xcodeproj/project.pbxproj:26]: ViewController is actually not used anywhere.

File: Navigation-Examples/Constants.swift [hunk 3]
```diff
@@ -12,7 +12,11 @@ typealias NamedController = (
 let listOfExamples: [NamedController] = [
     (
         name: "Advanced Implementation",
-        description: "Demonstrates how to display a custom map style and how to apply stylized components in the UI. Long press on the map to begin.",
+        description:"""
+        Demonstrates how to display a custom map style and how to apply stylized components in the UI.
+        Example also allows user to select an alternate route. Long press on the map to begin.
```
Comment (1ec5) [inline Navigation-Examples/Constants.swift:17]:
```suggestion
        This example also allows the user to select an alternate route. Long press on the map to begin.
```

File: Navigation-Examples/Constants.swift [hunk 4]
```diff
@@ -79,5 +76,19 @@ let listOfExamples: [NamedController] = [
...
+    (
+        name: "Route Deserialization",
+        description: "Demonstrates how to initialize a Route and deserialize it from JSON.",
+        controller: RouteDeserializationViewController.self,
+        storyboard: nil,
+        pushExampleToViewController: false
+    ),
+    (
+        name: "Route Initialization",
+        description: "Demonstrates how to initialize a Route using initialization API.",
```
Comment (1ec5) [inline Navigation-Examples/Constants.swift:89]:
```suggestion
        description: "Demonstrates how to initialize a Route using initializers in code.",
```

**Pull request merged**